app/models/event.rb

```diff
@@ -19,8 +19,7 @@ def self.total_capital_gain(year)
   end
 
   def self.events_for_year(year)
-    day = Date.new(year)
-    year_range = (day...day + 1.year)
+    year_range = Date.new(year).all_year
```

Comment:
Subtle difference but actually `. all_year` is more correct.

```ruby
year = 2021

day = Date.new(year)
=> Fri, 01 Jan 2021
(day...day + 1.year)
=> Fri, 01 Jan 2021...Sat, 01 Jan 2022

Date.new(2021).all_year
=> Fri, 01 Jan 2021..Fri, 31 Dec 2021
```

Reply:
Notice how one uses `...` and the other uses `..`, so I think the two ranges are actually equivalent.
```rb
year = 2021
day = Date.new(year)
#=> Fri, 01 Jan 2021

r1 = (day...day + 1.year)
#=> Fri, 01 Jan 2021...Sat, 01 Jan 2022
r1.include?(Date.parse "2021-12-31")
#=> true
r1.include?(Date.parse "2022-01-01")
#=> false

r2 = Date.new(2021).all_year
#=> Fri, 01 Jan 2021..Fri, 31 Dec 2021
r2.include?(Date.parse "2021-12-31")
#=> true
r2.include?(Date.parse "2022-01-01")
#=> false
```

The generated SQL reads better, though:
```rb
puts Event.where(executed_at: r1).to_sql
#=> SELECT "events".* FROM "events" WHERE "events"."executed_at" >= '2021-01-01' AND "events"."executed_at" < '2022-01-01'

puts Event.where(executed_at: r2).to_sql
#=> SELECT "events".* FROM "events" WHERE "events"."executed_at" BETWEEN '2021-01-01' AND '2021-12-31'
```